packages/core/vue/src/useWallet.ts

```diff
@@ -88,57 +89,67 @@ export const createWalletStore = ({
         wallet: Wallet | null;
         adapter: Adapter | null;
         publicKey: PublicKey | null;
-        ready: boolean;
         connected: boolean;
     }) => {
-        wallet.value = state.wallet;
+        wallet.value = state.wallet
+            ? {
+                  ...state.wallet,
+                  readyState: state.wallet.adapter.readyState,
+              }
+            : null;
         adapter.value = state.adapter;
-        ready.value = state.ready;
         publicKey.value = state.publicKey;
         connected.value = state.connected;
     };
     const resetState = () => {
         setState({
             wallet: null,
             adapter: null,
-            ready: false,
             publicKey: null,
             connected: false,
         });
     };
 
-    // Create a wallet dictionary keyed by their name.
-    const walletsByName = computed(() => {
-        return wallets.reduce<Record<WalletName, Wallet>>((walletsByName, wallet) => {
-            walletsByName[wallet.name] = wallet;
-            return walletsByName;
-        }, {});
+    // Wrap wallets to conform to the `WalletsWithReadyState` interface.
+    const walletsWithReadyState = computed(() =>
+        wallets.map((wallet) => ({
+            ...wallet,
+            readyState: wallet.adapter.readyState,
+        }))
+    );
+
+    // When the wallets change, start to listen for changes to their `readyState`.
+    watch(wallets, (_oldWallets, _newWallets, onInvalidate) => {
+        function handleReadyStateChange(this: Wallet, nextReadyState: WalletReadyState) {
+            // FIXME: Did not test this; don't know if this is the Vue-y way to achieve this.
```

Comment:
I have no idea if this works or not, or if any of this makes sense in Vue. I'd have to dive deeper.

Reply:
Because of the structural changes to the base package, I just reverted all the Angular, Svelte, and Vue changes for now and kept them on the current version of the base package. We can update them for consistency when we're confident in the API.